Source/Core/Core/HW/WiimoteEmu/Extension/BalanceBoard.h

```diff
@@ -0,0 +1,75 @@
+// Copyright 2010 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <array>
+
+#include "Core/HW/WiimoteCommon/WiimoteReport.h"
+#include "Core/HW/WiimoteEmu/Dynamics.h"
+#include "Core/HW/WiimoteEmu/Extension/Extension.h"
+
+namespace ControllerEmu
+{
+class AnalogStick;
+class Triggers;
+}  // namespace ControllerEmu
+
+namespace WiimoteEmu
+{
+enum class BalanceBoardGroup
+{
+  Balance,
+  Weight
+};
+
+class BalanceBoard : public Extension1stParty
+{
+public:
+  union DataFormat
+  {
+    struct
+    {
+      u16 top_right;
+      u16 bottom_right;
+      u16 top_left;
+      u16 bottom_left;
+      u8 unknown1;
+      u8 pad;
+      u8 battery;
+    };
+  };
+  static_assert(sizeof(DataFormat) == 12, "Wrong size");
+
+  BalanceBoard();
+
+  void Update() override;
+  bool IsButtonPressed() const override;
+  void Reset() override;
+  void DoState(PointerWrap& p) override;
+
+  ControllerEmu::ControlGroup* GetGroup(BalanceBoardGroup group);
+
+  void LoadDefaults(const ControllerInterface& ciface) override;
+
+  // Use the same calibration data for all sensors.
+  // Wii Fit appears to convert to this internally (so it doesn't care about differences smaller
+  // than a gram). Normal balance boards tend to be less precise, usually around 10 grams.
+  static constexpr u16 WEIGHT_0_KG = 0;
+  static constexpr u16 WEIGHT_17_KG = 17000;
+  static constexpr u16 WEIGHT_34_KG = 34000;
+
+private:
+  ControllerEmu::AnalogStick* m_balance;
+  ControllerEmu::Triggers* m_weight;
+
+  static constexpr u16 LOW_WEIGHT_DELTA = WEIGHT_17_KG - WEIGHT_0_KG;
```

Comment:
These 3 constants can be moved into the cpp file since nothing in this source file relies on them, they're purely implementation details.